network/test/middleware_test.go

```diff
@@ -353,8 +388,10 @@ func (m *MiddlewareTestSuite) TestUnsubscribe() {
 		require.NoError(m.Suite.T(), err)
 	}
 
-	// wait for nodes to form a mesh
-	time.Sleep(2 * time.Second)
+	// set up waiting for m.size pubsub tags indicating a mesh has formed
+	for i := 0; i < m.size; i++ {
+		<-m.obs
```

Comment:
```suggestion
                unittest.RequireCloseBefore(t, m.obs, 2*time.Second, "could not receive pubsub tag indicating mesh formed")
```
better not to allow blocking code on tests without timeout.